db/dcrpg/pgblockchain.go

```diff
@@ -4272,3 +4395,1479 @@ func (pgb *ChainDB) GenesisStamp() int64 {
+		NumFundingTxns:    numReceiving,
+		NumSpendingTxns:   numSpending,
+		AmountReceived:    totalreceived,
+		AmountSent:        totalsent,
+		AmountUnspent:     totalreceived - totalsent,
+		Balance:           balance,
+		KnownTransactions: numberMaxOfTx,
+		KnownFundingTxns:  numReceiving,
+		KnownSpendingTxns: numSpending,
+	}
+
+	// Sort by date and calculate block height.
+	addrData.PostProcess(uint32(pgb.Height()))
+
+	return addrData, addrType, nil
+}
+
+// GetTip grabs the highest block stored in the database.
+func (pgb *ChainDB) GetTip() (*exptypes.WebBasicBlock, error) {
+	tip, err := pgb.getTip()
+	if err != nil {
+		return nil, err
+	}
+	blockdata := exptypes.WebBasicBlock{
+		Height:      tip.Height,
+		Size:        tip.Size,
+		Hash:        tip.Hash,
+		Difficulty:  tip.Difficulty,
+		StakeDiff:   tip.StakeDiff,
+		Time:        tip.Time.S.UNIX(),
+		NumTx:       tip.NumTx,
+		PoolSize:    tip.PoolInfo.Size,
+		PoolValue:   tip.PoolInfo.Value,
+		PoolValAvg:  tip.PoolInfo.ValAvg,
+		PoolWinners: tip.PoolInfo.Winners,
+	}
+	return &blockdata, nil
+}
+
+// getTip returns the last block stored using StoreBlockSummary.
+// If no block has been stored yet, it returns the best block in the database.
+func (pgb *ChainDB) getTip() (*apitypes.BlockDataBasic, error) {
+	pgb.tipMtx.RLock()
+	defer pgb.tipMtx.RUnlock()
+	if pgb.tipSummary != nil && pgb.tipSummary.Hash == pgb.BestBlockHashStr() {
+		return pgb.tipSummary, nil
+	}
+	tip, err := RetrieveLatestBlockSummary(pgb.ctx, pgb.db)
+	if err != nil {
+		return nil, err
+	}
+	pgb.tipSummary = tip
+	return tip, nil
+}
+
+// DecodeRawTransaction creates a *chainjson.TxRawResult from a hex-encoded
+// transaction.
+func (pgb *ChainDB) DecodeRawTransaction(txhex string) (*chainjson.TxRawResult, error) {
+	bytes, err := hex.DecodeString(txhex)
+	if err != nil {
+		log.Errorf("DecodeRawTransaction failed: %v", err)
+		return nil, err
+	}
+	tx, err := pgb.Client.DecodeRawTransaction(bytes)
+	if err != nil {
+		log.Errorf("DecodeRawTransaction failed: %v", err)
+		return nil, err
+	}
+	return tx, nil
+}
+
+// TxHeight gives the block height of the transaction id specified
+func (pgb *ChainDB) TxHeight(txid *chainhash.Hash) (height int64) {
+	txraw, err := pgb.Client.GetRawTransactionVerbose(txid)
+	if err != nil {
+		log.Errorf("GetRawTransactionVerbose failed for: %v", txid)
+		return 0
+	}
+	height = txraw.BlockHeight
+	return
+}
+
+// GetExplorerFullBlocks gets the *exptypes.BlockInfo's for a range of block
+// heights.
+func (pgb *ChainDB) GetExplorerFullBlocks(start int, end int) []*exptypes.BlockInfo {
+	if start < end {
+		return nil
+	}
+	summaries := make([]*exptypes.BlockInfo, 0, start-end)
+	for i := start; i > end; i-- {
+		data := pgb.GetBlockVerbose(i, true)
+		block := new(exptypes.BlockInfo)
+		if data != nil {
+			block = pgb.GetExplorerBlock(data.Hash)
+		}
+		summaries = append(summaries, block)
+	}
+	return summaries
+}
+
+// Difficulty returns the difficulty.
```

Comment:
`Difficulty` vs. `RetreiveDifficulty` is a bit confusing, and now's a good time to clarify it I think since we're already making breaking changes.

Perhaps we can make `Difficulty` into `CurrentDifficulty` and update the comment to reflect that it's retrieved from dcrd.  And `RetreiveDifficulty` would be better off without the "Retrieve" part (esp. since it's spelled incorrectly).... plus the comment about 24hrs is super confusing.